tiling.js

```diff
@@ -4075,16 +4081,23 @@ export function insertWindow(metaWindow, options = {}) {
         }
     }
     if (spaceOfWindow !== space) {
-        console.log("#winprops", `Inserting window into space ${space.name}`);
+        console.log("#winprops", `Inserting window into space ${space.name} (current space ${spaceOfWindow.name})`);
         spaceOfWindow.removeWindow(metaWindow);
         metaWindow.foreach_transient(t => {
             spaceOfWindow.removeWindow(t);
         });
+        // TODO do we need metaWindow.move_to_monitor(...)
         metaWindow.change_workspace(space.workspace);
         metaWindow.foreach_transient(t => {
             space.addFloating(t);
         });
-        // activate?
+
+        if (focusOnOpen) {
+            console.log("#winprops", "focussing space of inserted window");
+            // TODO doesn't seem to work
+            spaces.animateToSpace(spaceOfWindow, space, true);
```

Comment:
Hey @Lythenas,

Instead of this, try using `callbackOnActorShow` approach (my guess is the window hasn't actually been shown yet when this is called).

Reply:
also note `animateToSpace` function signature is the other way around `animateToSpace(to, from, animate = true, callback)`.

That is you need spaceOfWindow to the second argument.

Reply:
anyway, that's more of an internal function (focused on animation).  You'd probably want something higher-up, like `spaces.switchWorkspace`.

I'll have a look at this when I can get some free time (which hasn't been of late).

Reply:
Hey @Lythenas, just pushed some changes to fix the selecting space.

Still getting a visual anomaly when switching spaces with these windows (so think hitting a corner case for window registration and clones).

Still it's progress.